src/reader.cpp

```diff
@@ -2774,6 +2782,7 @@ void reader_data_t::handle_readline_command(readline_cmd_t c, readline_loop_stat
                 }
                 rls.finished = true;
                 update_buff_pos(&command_line, command_line.size());
+                exec_prompt(L"after_input");
```

Comment:
This now executes the prompt twice which seems unnecessary - I think the previous one was better.

Alternatively, wouldn't prefixing the right prompt with `#` work in most cases?
Or maybe it's easy to implement something like a transient prompt in fish script.

Reply:
> This now executes the prompt twice which seems unnecessary - I think the previous one was better.

This change is more desirable than previous for me. Because this change can realize not only `TRANSIENT_RPROMPT` but also the prompt function I'm using in zsh.

With this change and following `fish_prompt` entry:
```fish
    if test $argv[1] = "before_input"
        set -g prompt_timestamp_before_input (date "+%H:%M:%S")
    end

    echo -n -s $arrow $cyan ~$prompt_timestamp_before_input ' ' (date "+%H:%M:%S")~ ' ' $repo_info $normal ' '
```
I can get prompts with timestamps of finished previous command and started command like:
```terminal.sh-session
➜ ~08:31:05 08:31:40~ (pass-input_pahse-parameter-to-prompt-functions) ✗ 
➜ ~08:31:40 08:35:12~ (pass-input_pahse-parameter-to-prompt-functions) ✗ sleep 10
➜ ~08:35:22 08:35:29~ (pass-input_pahse-parameter-to-prompt-functions) ✗ 
```

Reply:
Yeah but it's not desirable to execute the prompt twice by default.
For example, if a prompt errors, then the user will see duplicate error messages.

Advanced use cases like this should be possible to implement using fish
functions - this is more flexible than introducing specific settings.
Find below an implementation using event handlers (see `--on-event` in
`help function`).  It should do exactly what you want.

That being said I thought #6419 was okay although I think it makes more
sense not to have a right prompt in the first place.


```fish
set -g timestamp_before (date "+%H:%M:%S")

function fish_prompt
    echo -n -s $timestamp_before ' ' (date "+%H:%M:%S") '> '
end

function set-timestamp-postexec --on-event fish_postexec
    set -g timestamp_before (date "+%H:%M:%S")
end

# Repaint the prompt before executing a command to update to time stamp
# and clear the right prompt.
function repaint-prompt-preexec --on-event fish_preexec
    set -l prompt (fish_prompt)
    set -l prompt_length (string length -- $prompt)
    set -l command_length (string length -- $argv)
    set -l commandline_length (math $prompt_length + $command_length)
    set -l commandline_line_count (math \
        ceil \( $commandline_length / $COLUMNS \) \
        + (count (string match -r --all -- \n $argv)) / 2
    )
    # Move the cursor up, to the beginning of the line (\r) an repaint the prompt.
    printf (string repeat -Nn $commandline_line_count '\033[A')\r%s $prompt
    
    # Transient right prompt: if there is space for a right prompt, clear it.
    set -l right_prompt_length (string length -- (fish_right_prompt))
    set -l right_prompt_offset (math $COLUMNS - $right_prompt_length)
    if test $commandline_length -lt $right_prompt_offset
        printf \r(string repeat -Nn $right_prompt_offset '\033[C')'\033[K'
    end

    # Restore cursor position
    string repeat -Nn $commandline_line_count \n
end
```

Reply:
> Yeah but it's not desirable to execute the prompt twice by default.

Even in zsh, option setting is required to enable the prompt update before executing the command. However, the idea of adding a similar option to fish-shell was considered unacceptable and was not included in this PR.

> … It should do exactly what you want.

Hm, it might be better if there is a built-in command that redraws the prompt that can be called from within the `fish_preexec` event function. 🤔